test/integration/sitemap_controller_test.rb

```diff
@@ -8,6 +8,6 @@ class SitemapControllerTest < ActionDispatch::IntegrationTest
 
   test 'should return the right xml' do
     get '/sitemap.xml'
-    assert_equal response.body, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<urlset xmlns=\"http://www.sitemaps.org/schemas/sitemap/0.9\" xmlns:geo=\"http://www.google.com/geo/schemas/sitemap/1.0\" xmlns:image=\"http://www.google.com/schemas/sitemap-image/1.1\" xmlns:mobile=\"http://www.google.com/schemas/sitemap-mobile/1.0\" xmlns:news=\"http://www.google.com/schemas/sitemap-news/0.9\" xmlns:pagemap=\"http://www.google.com/schemas/sitemap-pagemap/1.0\" xmlns:video=\"http://www.google.com/schemas/sitemap-video/1.1\" xmlns:xhtml=\"http://www.w3.org/1999/xhtml\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd\">\n  <url>\n    <loc>localhost:3000</loc>\n    <lastmod>2016-09-22T18:11:05-03:00</lastmod>\n    <changefreq>always</changefreq>\n    <priority>1.0</priority>\n  </url>\n\n    <url>\n      <loc>localhost:3000/articles</loc>\n      <lastmod>2016-09-22T18:11:05-03:00</lastmod>\n      <changefreq>weekly</changefreq>\n      <priority>0.5</priority>\n    </url>\n</urlset>\n"
+    assert_equal response.body, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<urlset xmlns=\"http://www.sitemaps.org/schemas/sitemap/0.9\" xmlns:geo=\"http://www.google.com/geo/schemas/sitemap/1.0\" xmlns:image=\"http://www.google.com/schemas/sitemap-image/1.1\" xmlns:mobile=\"http://www.google.com/schemas/sitemap-mobile/1.0\" xmlns:news=\"http://www.google.com/schemas/sitemap-news/0.9\" xmlns:pagemap=\"http://www.google.com/schemas/sitemap-pagemap/1.0\" xmlns:video=\"http://www.google.com/schemas/sitemap-video/1.1\" xmlns:xhtml=\"http://www.w3.org/1999/xhtml\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd\">\n  <url>\n    <loc>http://www.example.com</loc>\n    <lastmod>2016-09-22T18:11:05-03:00</lastmod>\n    <changefreq>always</changefreq>\n    <priority>1.0</priority>\n  </url>\n\n    <url>\n      <loc>http://www.example.com/articles</loc>\n      <lastmod>2016-09-22T18:11:05-03:00</lastmod>\n      <changefreq>weekly</changefreq>\n      <priority>0.5</priority>\n    </url>\n</urlset>\n"
```

Comment:
I would save this xml to a subfolder and read the file and assert against it content. So the test looks cleaner and is easier to maintain